### PR TITLE
tests: Move arith tests to MLIR syntax

### PR DESCRIPTION
**Pull request by math-fehr**
Also, use the test dialect to make it more concise

### CODE REVIEW

File: tests/filecheck/dialects/arith/arith_attrs.mlir [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// RUN: xdsl-opt %s -t mlir | xdsl-opt -f mlir -t mlir | filecheck %s
```
Comment (tobiasgrosser) [inline tests/filecheck/dialects/arith/arith_attrs.mlir:1]: Do we need the `-t mlir` here? Or is this already the default?

**Review by tobiasgrosser — APPROVED** [review]

File: tests/filecheck/dialects/arith/arith_attrs.mlir (hunk 1 above)
Comment (tobiasgrosser) [reply tests/filecheck/dialects/arith/arith_attrs.mlir:1]: I see that it's not yet the default. I guess we first switch all the tests and then change the default.

**Review by superlopuh — APPROVED** [review]

**Review by georgebisbas — COMMENTED** [review]
So If I undersand correctly we drop the conversion tests of xdsl syntax. Why is that?

**Comment by webmiche** [comment]
The xDSL syntax will be dropped in the near future. We are moving everything to MLIR syntax.

**Pull request merged**